### ISSUE TITLE
changepassword method always return "We've just sent you an email to reset your password."

### ISSUE THREAD

**Issue by VaizardLLL** [auth0/auth0.js#380]
Hi, there!

First, I'm using version *http://cdn.auth0.com/js/auth0/8.3.0/auth0.min.js* but I also got the same error with version *8.0.0*. I'm also using Angular2 with my SPA

Every time that I use the function **changepassword**, I got the same response: **"We've just sent you an email to reset your password."** even when the email doesn't exist. If I consult the auth0 portal log, I can see the error **"User does not exist."**

If I send a correct email, it's all good. I got the email with the link to change the password.

Here is the code I'm using:

```typescript
public changePassword(email: string) {
    this.auth0.changePassword({
      connection: 'MyConn',
      email: email
    }, (err, resp) => {
      if (err) {
        console.log(err.message);
      }
      else {
        console.log(resp);
      }
    });
  }
```

**Comment by hzalaz** [comment]
Its the intended behaviour @VaizardLLL

**Comment by VaizardLLL** [comment]
Hum, maybe I don't make myself clear... let me try again:

 I'm trying to recover a password from a user (email) that it's not in my database (connection on auth0). Then I'm getting the info through the auth0-js as "We've just sent you an email to reset your password.". 

**This is wrong because the email/user do not exist!**

If I look the log of this call at the auth0 portal log, I can see the correct reply "The user do no exist."

So this case, that I'm explaining now is an intended behavior?

Thanks for reply @hzalaz

**Comment by hzalaz** [comment]
It's the intended behaviour because if you check the network tab its what the API answers 😄 

```
POST /dbconnections/change_password HTTP/1.1
Accept: application/json
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 133
Content-Type: application/json
Host: tenant.auth0.com
User-Agent: HTTPie/0.9.2

{
    "client_id": "...", 
    "connection": "Username-Password-Authentication", 
    "email": "not-found@example.xom"
}

HTTP/1.1 200 OK
Connection: keep-alive
Content-Length: 54
Content-Type: application/json; charset=utf-8
Date: Thu, 09 Mar 2017 19:06:54 GMT
Keep-Alive: timeout=100
Strict-Transport-Security: max-age=15724800
X-RateLimit-Limit: 10
X-RateLimit-Remaining: 9
X-RateLimit-Reset: 1489086475
X-Robots-Tag: noindex, nofollow, nosnippet, noarchive
cache-control: no-cache
strict-transport-security: max-age=15724800
x-auth0-requestid: 9fb8b28fe7306e9c5557
x-powered-by: Express
x-robots-tag: noindex, nofollow, nosnippet, noarchive

"We've just sent you an email to reset your password."
````

**Comment by VaizardLLL** [comment]
Ow.... I think I will have to make a workaround and check my app database first.

Thanks again @hzalaz for the reply!

**Comment by luisrudge** [comment]
@VaizardLLL This is by default and follows security best practices. Please check this [blog post](https://www.troyhunt.com/everything-you-ever-wanted-to-know/) - specially the part about "Username enumeration and the impact on anonymity".

**Comment by VaizardLLL** [comment]
@luisrudge Yeap, just think about it now and make sense. Thanks for the link/info :)
